### PR TITLE
refactor: remove moac related changes 

### PR DESCRIPTION
**Pull request by rohan2794**
JIRA: https://mayadata.atlassian.net/browse/MQ-2729

### CODE REVIEW

File: src/MQ-2330-ms_pod_disruption_rm_vol/replica_pod_remove_rm_vol_test.go [hunk 1]
```diff
@@ -318,27 +318,7 @@ func ReplicaLossVolumeDelete(pvcName string, storageClassName string, fioPodName
...
-		// verify that the replicas do not get removed
-		logf.Log.Info("wait for 30s before rechecking replicas")
-		time.Sleep(time.Duration(30) * time.Second)
-
-		for _, nodeIP := range replicaIPs {
-			reps = listReplicasOnNode(nodeIP)
-			Expect(len(reps)).To(Equal(1), "Expected 1 replica on each replica pod")
-		}
-	} else {
+	if controlplane.MajorVersion() == 1 {
```
Comment (chriswldenyer) [inline src/MQ-2330-ms_pod_disruption_rm_vol/replica_pod_remove_rm_vol_test.go:321]: We should probably fail the test if controlplane.MajorVersion() != 1, if we want to keep controlplane.MajorVersion() for future use.

File: src/common/k8sinstall/util.go [hunk 2]
```diff
@@ -137,10 +132,6 @@ func WaitForPoolCrd() bool {
 func GenerateControlPlaneYamlFiles() error {
 	e2eCfg := e2e_config.GetConfig()
 
-	if controlplane.MajorVersion() == 0 {
-		return nil
-	}
-
 	if controlplane.MajorVersion() == 1 {
```
Comment (chriswldenyer) [inline src/common/k8sinstall/util.go:135]: else fail if not 1 ?

File: src/common/k8stest/util.go [hunk 3]
```diff
@@ -723,15 +678,10 @@ func CheckAndSetControlPlane() error {
...
-	if !foundMoac && !foundCoreAgents {
-		return fmt.Errorf("MOAC and Restful Control plane components are absent")
-	}
-	if foundMoac {
-		version = "0.8.2"
+	if !foundCoreAgents {
+		return fmt.Errorf("restful Control plane components are absent")
 	}
+
 	if foundCoreAgents {
```
Comment (chriswldenyer) [inline src/common/k8stest/util.go:685]: The "if" is redundant here.

File: src/common/k8stest/util_pvc.go [hunk 4]
```diff
@@ -202,9 +201,6 @@ func MkPVC(volSizeMb int, volName string, scName string, volType common.VolumeTy
 // MsvConsistencyCheck check consistency of  MSV Spec, Status, and associated objects returned by gRPC
 func MsvConsistencyCheck(uuid string) error {
 	//FIXME: implement new MsvConsistencyCheck inline with mayastor control plane
-	if controlplane.MajorVersion() != 0 {
```
Comment (chriswldenyer) [inline src/common/k8stest/util_pvc.go:205]: So this wasn't running with the MCP? Looks like a behaviour change with this PR, so we should check it.

**Review by chriswldenyer — COMMENTED** [review]
Looks good, but I think in all cases where we do "if controlplane.MajorVersion() == 1", (or a switch statement based on the value) we should fail if it is not 1 (or is case default). Some switch statements do this and some don't, so we should be consistent IMO.

File: src/common/k8stest/util_pvc.go (hunk 4 above)
Comment (blaisedias) [reply src/common/k8stest/util_pvc.go:205]: Indeed it was disabled - we need to look into adding it. See the FIXME.
Now that release pressure is off we can try enabling consistency checks.
Last time I looked it was tricky avoid false posistives :-(

File: src/tools/extended-test-framework/common/constants.go [hunk 5]
```diff
@@ -49,7 +49,6 @@ const MayastorEngineLabelValue = "mayastor"
 const FsxBlockFileName = "/dev/sdm"
 
 // Control plane identifiers for config
-const CpMoac = "moac"
 const CpMcp2 = "mcp2"
```
Comment (blaisedias) [inline src/tools/extended-test-framework/common/constants.go:52]: I think we can remove CpMcp2 as well - I cannot find any uses for it.

File: src/common/k8sinstall/util.go (hunk 2 above)
Comment (rohan2794) [reply src/common/k8sinstall/util.go:135]: we are returning error if control plane version is not 1
https://github.com/mayadata-io/mayastor-e2e/pull/418/files#diff-8fce4e3b2d6362c58e3f73dccf4187c09bb948bbcf8a5ad89df6bb97d7da47f0R161

File: src/common/k8stest/util.go (hunk 3 above)
Comment (rohan2794) [reply src/common/k8stest/util.go:685]: good point

File: src/tools/extended-test-framework/common/constants.go (hunk 5 above)
Comment (chriswldenyer) [reply src/tools/extended-test-framework/common/constants.go:52]: I am happy for you to make this change. There are other changes to the extended framework to remove moac but I can do those in a separate PR.

**Review by chriswldenyer — APPROVED** [review]
LGTM

**Review by blaisedias — APPROVED** [review]

File: src/tools/extended-test-framework/common/constants.go (hunk 5 above)
Comment (rohan2794) [reply src/tools/extended-test-framework/common/constants.go:52]: ok, thanks

**Comment by rohan2794** [comment]
bors merge

**Comment by rohan2794** [comment]
bors merge

**Pull request merged**